src/extensions/default/Inline3DParametersEditor/ParameterEditor.js

```diff
@@ -0,0 +1,214 @@
+            this.$sliders.push(this.$element.find("#input-"+(i+1)));
+        }
+    };
+
+    /**
+     * Initialises the _spaces array to store number of spaces between parameters
+     */
+    ParameterEditor.prototype._setLabels = function() {
+        var parameter = JSON.parse(Parameters)[this._tag];
+        if(!parameter) {
+            return;
+        }
+        var labels = parameter.labels;
+        if(!labels) {
+            return;
+        }
+        for(var i = 0; i < labels.length; i++) {
+            this.$element.find("#label-"+(i+1)).html(labels[i] + " &rarr;");
+        }
+    };
+
+    /*
+     * Returns the number of parameters based on the parameters string
+     * @param {parameters} String consisting of parameters
+     * @return {Number}
+     */
+    ParameterEditor.prototype._getNumberOfProperties = function(parameters) {
+        var parametersArray = parameters.trim().match(/\S+/g);
+        return parametersArray.length;
+    };
+
+    /*
+     * Returns a modified offset to smoothen the scroll on the basis of the
+     * modifier of the ParameterEditor object.
+     */
+    ParameterEditor.prototype._getOffset = function(pos, zeroPos) {
+        var offset = pos - zeroPos;
+        return offset / this._modifier;
+    };
+
+    ParameterEditor.prototype._handleSliderDrag = function(event) {
+        var index = event.data.index;
+        var xPos = this._position;
+        var offset = this._getOffset(event.clientX, xPos);
+        var n = this._values[index] + offset;
+        this.$sliders[index].val(n.toFixed(2));
+        this._commitParameters(this._getParameters());
+    };
+
+    /*
+     * Returns a string consisting of as many space characters as the
+     * inptt parameter
+     * @param {number} Number of spaces required
+     * @return {String}
+     */
+    ParameterEditor.prototype._getWhiteSpaces = function(number) {
+        var spaces = "";
+        for(var i = 0 ; i < number; i++) {
+            spaces += " ";
+        }
+        return spaces;
+    };
+
+    ParameterEditor.prototype._getParameters = function() {
+        var parameters = "";
+        for(var i = 0; i < this._numberOfParameters-1; i++) {
+            parameters += this.$sliders[i].val();
+            parameters += this._getWhiteSpaces(this._spaces[i]);
+        }
+        parameters += this.$sliders[i].val();
+        return parameters;
+    };
+
+    ParameterEditor.prototype._registerDragHandler = function ($element, handler, index) {
+        var mouseupHandler = function (event) {
+            $(window).unbind("mousemove", handler);
+            $(window).unbind("mouseup", mouseupHandler);
+        };
+        $element.mousedown(function (event) {
+            $(window).bind("mousemove", {"index" : index}, handler);
+            $(window).bind("mouseup", mouseupHandler);
+        });
+        $element.mousedown({"index" : index}, handler);
+    };
+
+    $(window.document).on("mousedown", function (e) {
```

Comment:
This might be right/necessary, but it seems heavy-handed to me to do this in `window.document`.  Can we not isolate this to some element closer to where we are working?